plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt

```diff
@@ -134,40 +137,66 @@ class GradleEnclavePlugin @Inject constructor(private val layout: ProjectLayout)
         )
     }
 
-    private fun buildUnsignedGramineEnclaveTask(target: Project, type: BuildType): BuildUnsignedGramineEnclave {
+    private fun generateGramineEnclaveMetadataTask(
+            target: Project,
+            type: BuildType,
+            conclaveExtension: ConclaveExtension,
+            signingKey: Provider<RegularFile?>
+    ): GenerateGramineEnclaveMetadata {
         val gramineBuildDir = baseDirectory.resolve("gramine").toString()
+
+        return target.createTask("generateGramineEnclaveMetadata$type") { task ->
+            task.buildType.set(type)
+            task.maxThreads.set(conclaveExtension.maxThreads.get())
+            task.signingKey.set(signingKey)
+            task.outputGramineEnclaveMetadata.set(
+                Paths.get(gramineBuildDir).resolve(GenerateGramineEnclaveMetadata.METADATA_FILE_NAME).toFile()
+            )
+        }
+    }
+
+    private fun buildUnsignedGramineEnclaveTask(
+            target: Project,
+            type: BuildType,
+            conclaveExtension: ConclaveExtension
+    ): BuildUnsignedGramineEnclave {
+        val gramineBuildDir = baseDirectory.resolve("gramine").toString()
```

Comment:
Can we move this out into a field or val, since the location is now used in multiple places, perhaps alongside `baseDirectory`.